### PR TITLE
ANDROID-4777 Added ipv6 filtering

### PR DESCRIPTION
**Pull request by vandac**
Inspired by https://github.com/instacart/truetime-android/pull/110

### CODE REVIEW

File: library-extension-rx/src/main/java/com/instacart/library/truetime/TrueTimeRx.java [hunk 1]
```diff
@@ -180,7 +201,23 @@ public void accept(long[] ntpResponse) {
...
+                                      for (InetAddress i : addresses) {
+                                          try {
+                                              if (filter.test(i)) {
+                                                  validAddresses.add(i);
+                                              }
+                                          } catch (Exception e) {
+                                              return Flowable.error(e);
+                                          }
+                                      }
+                                      return Flowable.fromIterable(validAddresses);
```
Comment (jan-ehrenberger-wandera) [inline library-extension-rx/src/main/java/com/instacart/library/truetime/TrueTimeRx.java:219]: Just wondering - what's the background behind using `Flowable.fromIterable` here and `Flowable.fromArray` above?

**Review by jan-ehrenberger-wandera — APPROVED** [review]

File: library-extension-rx/src/main/java/com/instacart/library/truetime/TrueTimeRx.java (hunk 1 above)
Comment (vandac) [reply library-extension-rx/src/main/java/com/instacart/library/truetime/TrueTimeRx.java:219]: addresses are stored in an array, whereas validAddresses are represented by a List (Iterable), so I believe that's what makes the difference

File: library-extension-rx/src/main/java/com/instacart/library/truetime/TrueTimeRx.java [hunk 2]
```diff
@@ -12,22 +25,15 @@
...
-import java.util.List;
-import org.reactivestreams.Publisher;
 
 public class TrueTimeRx
       extends TrueTime {
 
     private static final TrueTimeRx RX_INSTANCE = new TrueTimeRx();
     private static final String TAG = TrueTimeRx.class.getSimpleName();
 
+    private Predicate<InetAddress> _addressFilter = null;
```
Comment (tstolarik) [inline library-extension-rx/src/main/java/com/instacart/library/truetime/TrueTimeRx.java:35]: why the underscore in the name? I don't think we use it anywhere else

**Review by tstolarik — APPROVED** [review]

File: library-extension-rx/src/main/java/com/instacart/library/truetime/TrueTimeRx.java (hunk 2 above)
Comment (vandac) [reply library-extension-rx/src/main/java/com/instacart/library/truetime/TrueTimeRx.java:35]: that's the way the fields in this library are marked

**Pull request merged**